server.go

```diff
@@ -15,9 +15,21 @@ type Server struct {
 	// Optional function that runs on error. Can be used for logging.
 	OnError func(error)
 
+	// Optional function called with every incoming connection.
+	OnConn func(*http.Request)
```

Comment:
If you pass a pointer here then the request can be modified by the user before it is passed to your handler. Is that what we want?